### ISSUE TITLE
StackOverflowError in jl_has_concrete_subtype using Tuple{Vararg}

### ISSUE THREAD

**Issue by luchr** [JuliaLang/julia#32483]
With this minimal example

```julia
struct TreePart
    children :: Tuple{Vararg{TreePart}}
end

println(TreePart( () ))
```
there is a inifinite loop in `jl_has_concrete_subtype`:
```
Internal error: encountered unexpected error in runtime:
StackOverflowError()
jl_has_concrete_subtype at /buildworker/worker/package_linux64/build/src/gf.c:2559
jl_has_concrete_subtype at /buildworker/worker/package_linux64/build/src/gf.c:2575
jl_has_concrete_subtype at /buildworker/worker/package_linux64/build/src/gf.c:2575
jl_has_concrete_subtype at /buildworker/worker/package_linux64/build/src/gf.c:2575
...
```
Julia-Version:
```
Julia Version 1.2.0-rc1.0
Commit 7097799cf1 (2019-05-30 02:22 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)
```
This also happens for 
```
Julia Version 1.3.0-DEV.373
Commit ad42d5de18 (2019-06-07 22:19 UTC)
Platform Info:
  OS: Linux (x86_64-redhat-linux)
  CPU: Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)
```

**Comment by luchr** [comment]
Sorry for the inconvenience. It works for the current master:
```
Julia Version 1.3.0-DEV.488
Commit 49fc01aa12 (2019-07-02 21:12 UTC)
Platform Info:
  OS: Linux (x86_64-redhat-linux)
  CPU: Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)

```

**Comment by JeffBezanson** [comment]
Probably fixed by #31064 .
